backend/src/components/permissionUtils.js

```diff
@@ -145,7 +145,9 @@ function checkSdcSchoolCollectionAccess(req, res, next) {
       message: 'The requested SDC School Collection was not found in the request.'
     });
   }
-  if (req.session.activeInstituteType !== 'SCHOOL' || res.locals.requestedSdcSchoolCollection.schoolID !== req.session.activeInstituteIdentifier) {
+  if(req.session.activeInstituteType === 'DISTRICT' && doesSchoolBelongToDistrict(res.locals.requestedSdcSchoolCollection.schoolID, req.session.activeInstituteIdentifier)) {
```

Comment:
I'm wondering if the `edxUserHasAccessToInstitute` function could work here as part of the if condition. What are your thoughts on this?